File: ShadowFrameworkLite20/src/shadow/pipeline/SFStructuresTable.java
package shadow.pipeline;

import shadow.renderer.SFAsset;
import shadow.system.data.SFDataset;
import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;

public class SFStructuresTable extends SFAsset {

	private SFStructureArray array;
	private SFPipelineStructureInstance structure;

	@Override
	public void allocateBuffers() {
		array=SFPipeline.getSfPipelineMemory().generateStructureData(structure);
	}

	public SFStructureArray getDataArray() {
		return array;
	}

	@Override
	public void deallocateBuffers() {
		// TODO Deallocation is still not supported
	}

	@Override
	public void readFromStream(SFInputStream stream) {
		//TODO
		//structure=stream.readString();
	}

	@Override
	public void writeOnStream(SFOutputStream stream) {
		//TODO
		//stream.writeString(structure);
	}

	@Override
	public SFDataset generateNewDatasetInstance() {
		return new SFStructuresTable();
	}

	@Override
	public String getCode() {
		return getClass().getSimpleName();
	}

//	public String getStructure() {
//		return structure;
//	}

}

File: trunk/ShadowFrameworkJS/SF20Core/src/shadow/pipeline/SFStructuresTable.js

function SFStructuresTable(){
}

SFStructuresTable.prototype = {

	allocateBuffers:function(){
		array  = SFPipeline.getSfPipelineMemory().generateStructureData(structure);
	},

	getDataArray:function(){
		return ,array;
	},

	deallocateBuffers:function(){
	// TODO Deallocation is still not supported;//Warning: Not well Identified 
	},

	readFromStream:function(stream){
	//TODO		//structure=stream.readString();//Warning: Not well Identified 
	},

	writeOnStream:function(stream){
	//TODO		//stream.writeString(structure);//Warning: Not well Identified 
	},

	generateNewDatasetInstance:function(){
		return ,new ,SFStructuresTable();
	},

	getCode:function(){
		return ,getClass().getSimpleName();
	}

};